source/Server/AdoClients/AdoApiClient.cs

```diff
@@ -57,96 +62,102 @@ internal string GetPersonalAccessToken(AdoUrl adoUrl)
+                return ResultFromExtension<(string title, int? commentCount)>.Failed($"Error while fetching work item details from Azure DevOps: {status.ToDescription(jObject, testing)}");
             }
 
             try
             {
-                var fields = jObject["fields"];
-                return (fields["System.Title"].ToString(), fields["System.CommentCount"]?.Value<int>());
+                var fields = jObject?["fields"];
+                if (fields == null)
+                    return ResultFromExtension<(string title, int? commentCount)>.Failed("Unable to interpret work item details from Azure DevOps. `fields` element is missing.");
+
+                return ResultFromExtension<(string title, int? commentCount)>.Success((fields?["System.Title"]?.ToString() ?? string.Empty, fields?["System.CommentCount"]?.Value<int>() ?? default(int)));
             }
             catch
             {
-                return SuccessOrErrorResult.Failure("Unable to interpret work item details from Azure DevOps.");
+                return ResultFromExtension<(string title, int? commentCount)>.Failed("Unable to interpret work item details from Azure DevOps.");
             }
         }
 
         /// <returns>Up to 200 comments on the specified work item.</returns>
-        public SuccessOrErrorResult<string[]> GetWorkItemComments(AdoProjectUrls adoProjectUrls, int workItemId)
+        public ResultFromExtension<string[]> GetWorkItemComments(AdoProjectUrls adoProjectUrls, int workItemId)
         {
             // ReSharper disable once StringLiteralTypo
             var (status, jObject) = client.Get($"{adoProjectUrls.ProjectUrl}/_apis/wit/workitems/{workItemId}/comments?api-version=4.1-preview.2",
                 GetPersonalAccessToken(adoProjectUrls));
 
             if (status.HttpStatusCode == HttpStatusCode.NotFound)
             {
-                return new string[0];
+                return ResultFromExtension<string[]>.Success(new string[0]);
             }
 
             if (!status.IsSuccessStatusCode())
             {
-                return SuccessOrErrorResult.Failure($"Error while fetching work item comments from Azure DevOps: {status.ToDescription(jObject)}");
+                return ResultFromExtension<string[]>.Failed($"Error while fetching work item comments from Azure DevOps: {status.ToDescription(jObject)}");
             }
 
             string[] commentsHtml;
             try
             {
-                commentsHtml = jObject["comments"]
-                    .Select(c => c["text"].ToString())
-                    .ToArray();
+                commentsHtml = jObject?["comments"]?
+                    .Select(c => c["text"]?.ToString())
+                    .Where(c => c != null)
+                    .Cast<string>() // cast to keep the compiler happy with nullable checks
```

Comment:
Ran into this on the server as well. Should we add an extension method to utils for this?